pkg/sources/github/receive_adapter/receive_adapter.go

```diff
@@ -98,30 +119,35 @@ func main() {
 	}
 }
 
-func postMessage(target string, m *github.PullRequestPayload) error {
-	jsonStr, err := json.Marshal(m)
-	if err != nil {
-		log.Printf("Error: Failed to marshal the message: %+v : %v", m, err)
-		return err
-	}
-
+func postMessage(target string, payload interface{}, source, eventType, eventID string) error {
 	URL := fmt.Sprintf("http://%s/", target)
-	req, err := http.NewRequest("POST", URL, bytes.NewBuffer(jsonStr))
+
+	ctx := event.EventContext{
+		CloudEventsVersion: event.CloudEventsVersion,
+		EventType:          eventType,
+		EventID:            eventID,
+		EventTime:          time.Now(),
+		Source:             source,
+	}
+	req, err := event.Binary.NewRequest(URL, payload, ctx)
 	if err != nil {
-		log.Printf("Error: Failed to create http request: %v", err)
+		log.Printf("Failed to marshal the message: %+v : %s", payload, err)
 		return err
 	}
 
-	req.Header.Set("Content-Type", "application/json")
+	log.Printf("Posting to %q", URL)
 	client := &http.Client{}
 	resp, err := client.Do(req)
 	if err != nil {
-		log.Printf("Error: Failed to do POST: %v", err)
 		return err
 	}
 	defer resp.Body.Close()
-	log.Printf("Error: response Status: %s", resp.Status)
-	body, _ := ioutil.ReadAll(resp.Body)
-	log.Printf("Error: response Body: %s", string(body))
+
+	if resp.StatusCode != http.StatusOK {
+		// TODO: in general, receive adapters may have to be able to retry for error cases.
+		log.Printf("response Status: %s", resp.Status)
+		body, _ := ioutil.ReadAll(resp.Body)
+		log.Printf("response Body: %s", string(body))
```

Comment:
If this is a 500, should we return an error?

Reply:
/idk

There are some questions here that are bigger than this PR around what the Receive Adapter should do if it gets an error from the "action". Should it try again? How many times?...